app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt

```diff
@@ -0,0 +1,420 @@
+    setUpTestApplicationComponent()
+    testCoroutineDispatchers.registerIdlingResource()
+  }
+
+  @After
+  fun tearDown() {
+    testCoroutineDispatchers.unregisterIdlingResource()
+    Intents.release()
+  }
+
+  @Test
+  fun testFragment_headerTextIsDisplayed() {
+    launchOnboardingProfileTypeActivity().use {
+      onView(withId(R.id.profile_type_title))
+        .check(matches(isDisplayed()))
+      onView(withId(R.id.profile_type_title))
+        .check(
+          matches(
+            withText(
+              R.string.onboarding_profile_type_activity_header
+            )
+          )
+        )
+    }
+  }
+
+  @Test
+  fun testFragment_navigationCardsAreDisplayed() {
+    launchOnboardingProfileTypeActivity().use {
+      onView(withId(R.id.profile_type_learner_navigation_card))
+        .check(matches(isDisplayed()))
+      onView(withId(R.id.profile_type_learner_navigation_card))
+        .check(
+          matches(
+            hasDescendant(
+              withText(R.string.onboarding_profile_type_activity_student_text)
+            )
+          )
+        )
+
+      onView(withId(R.id.profile_type_supervisor_navigation_card))
+        .check(matches(isDisplayed()))
+      onView(withId(R.id.profile_type_supervisor_navigation_card))
+        .check(
+          matches(
+            hasDescendant(
+              withText(R.string.onboarding_profile_type_activity_parent_text)
+            )
+          )
+        )
+    }
+  }
+
+  @Test
+  fun testFragment_portrait_stepCountTextIsDisplayed() {
+    launchOnboardingProfileTypeActivity().use {
+      onView(withId(R.id.onboarding_steps_count))
+        .check(matches(isDisplayed()))
+      onView(withId(R.id.onboarding_steps_count))
+        .check(matches(withText(R.string.onboarding_step_count_two)))
+    }
+  }
+
+  @RunOn(TestPlatform.ESPRESSO) // Robolectric is usually not used to test the interaction of
+  // Android components
+  @Test
+  fun testFragment_backButtonClicked_currentScreenIsDestroyed() {
+    launchOnboardingProfileTypeActivity().use { scenario ->
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.onboarding_navigation_back)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      if (scenario != null) {
+        assertThat(scenario.state).isEqualTo(Lifecycle.State.DESTROYED)
+      }
+    }
+  }
+
+  @RunOn(TestPlatform.ESPRESSO) // Robolectric is usually not used to test the interaction of
+  // Android components
+  @Test
+  fun testFragment_landscapeMode_backButtonClicked_currentScreenIsDestroyed() {
+    launchOnboardingProfileTypeActivity().use { scenario ->
+      onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.onboarding_navigation_back))
+        .perform(click())
+      testCoroutineDispatchers.runCurrent()
+      if (scenario != null) {
+        assertThat(scenario.state).isEqualTo(Lifecycle.State.DESTROYED)
+      }
+    }
+  }
+
+  @Test
```

Comment:
I am not sure if every test should annotate the framework it is using

Reply:
If it is not annotated with the patform, they will run on both. If it is annotated, it means that it will only run on the specified platform, mainly because there is some sort of issue with running on the alternate platform.